### ISSUE TITLE
zipfile append mode for decks involving large media files

### ISSUE THREAD

**Issue by yash-fn** [kerrickstaley/genanki#103]
TL;DR: 
Change `'w'` to `'a'` for zipfile write part of code.

While developing my own decks I often tweak many things for the way I want my cards generated. When involving large media writes (50gb plus), it takes long time to rewrite entire media plus sqlite db when only sqlite db is what changed. plus wear on ssd although minor concern to time and efficiency of updates. Perhaps the highlighted line below should be changed to append ('a' mode for zipfile module) to only overwrite changed files? If someone has a deck already at the prescribed location perhaps they often are only updating something rather creating entirely new deck. If they really truly want entire new deck on each write they can specify a tempfile and then overwrite to permanent file once complete. This way genanki can support easily modifying already created decks when minor edits are involved. Thoughts?

https://github.com/kerrickstaley/genanki/blob/5026448cb661570b2355afc5a45c1c9fcc9eea24/genanki/package.py#L45

**Comment by yash-fn** [comment]
please disregard previous edit suggestion. It does not do what I expect. Apparently you can append the same file to the archive and instead of overwriting it it just keeps all versions of the file. So creating a new file is only way to ensure no duplicate. But I still think we need to find some way to not keep overwriting same large media files. I will look into this and see if I find something.

**Comment by yash-fn** [comment]
Okay, from what I can gather, not possible. You can add new files, but you can't delete and therefore can't overwrite any zip contents. But it appears anki can import a package of just media files which a user can just do on their own. So just gonna close this.

**Comment by yash-fn** [comment]
Actually, wait. What if we externalize the `with zipfile.ZipFile(file, 'w') as outzip:` into a separate function that the user can then overwrite? Call it something like write_function. That way a user can handle the zipfile part however they want but still benefit from genanki creating that sqlite db for them? Thoughts? This, for example would help cases like: #102
